pkgs/applications/networking/mailreaders/neomutt/default.nix

```diff
@@ -73,42 +73,51 @@ stdenv.mkDerivation rec {
   ''
   # https://github.com/neomutt/neomutt-contrib
   # Contains vim-keys, keybindings presets and more.
-  + lib.optionalString withContrib "${lib.getExe lndir} ${passthru.contrib} $out/share/doc/neomutt";
+  + lib.optionalString withContrib "${lib.getExe lndir} ${finalAttrs.passthru.contrib} $out/share/doc/neomutt";
 
   doCheck = true;
 
+  disabledTests = [
+    # Fails with node_padding.c:135: Check rc == 15... failed
+    "test_expando_node_padding"
+  ];
+
   preCheck = ''
-    cp -r ${passthru.test-files} $(pwd)/test-files
+    cp -r ${finalAttrs.passthru.test-files} $(pwd)/test-files
+
     chmod -R +w test-files
     (cd test-files && ./setup.sh)
 
     export NEOMUTT_TEST_DIR=$(pwd)/test-files
+
+    # Remove disabled tests from test runner
+    sed -i '/(${builtins.concatStringsSep "|" finalAttrs.disabledTests})/d' test/main.c
```

Comment:
`sed` should be avoided. you may use `substituteInPIace` instead

Reply:
Thank you, @nim65s, for taking the time to have another look at this and providing feedback.

What are the reasons that `sed` should be avoided? If I'm not mistaken it is part of `stdenv`…

I thought about using `substituteInPIace` at afirst, but wanted to delete the line(s) matching a regex instead of substituting matching text as the neovim tests are defined as a preprocessor `define` that spans multiple lines with line continuation (see [`test/main.c:313f` from neomutt/neomutt](https://github.com/neomutt/neomutt/blob/main/test/main.c#L313)).

My approach to use `substituteInPlace` would be as follows:

```sh
    for test in ${builtins.concatStringsSep " " finalAttrs.disabledTests}; do
      substituteInPlace test/main.c --replace-fail "NEOMUTT_TEST_ITEM($test)" ""
    done
```

What I dislike about this approach is that it calls `substituteInPlace` for every disabled test 🙁 Is there a more nixpkgsy or elegant way?

Reply:
> What are the reasons that `sed` should be avoided? If I'm not mistaken it is part of `stdenv`…

It doesn't log none matches.

> What I dislike about this approach is that it calls substituteInPlace for every disabled test 🙁 Is there a more nixpkgsy or elegant way?

You can call it with multiple --replace-* args

Reply:
How could multiple `--replace-*` args be constructed from `disabledTests`?

The question might make less sense for this package, as only a single test needs to be disabled, yet I'm curious about features/capabilites/uses of Nix that I may not be aware of 🙂

Reply:
I went ahead with the most pragmatic solution, if there is interest and appropriate feedback I'll happily work that in to make disabled failing tests for this package "smarter"…